packages/grid_client/src/modules/tft.ts

```diff
@@ -0,0 +1,53 @@
+import { QueryClient } from "@threefold/tfchain_client";
+import Decimal from "decimal.js";
+
+import { expose, validateInput } from "../helpers";
+import { calculator as Calculator } from "./calculator";
+import { TFTModel, USDModel } from "./models";
+
+enum substrateURL {
+  DEV = "wss://tfchain.dev.grid.tf/ws",
+  QA = "wss://tfchain.qa.grid.tf/ws",
+  TEST = "wss://tfchain.test.grid.tf/ws",
+  MAIN = "wss://tfchain.grid.tf/ws",
+}
+
+class TFT {
+  public SUBSTRATE_URL: string;
+  private calculator: Calculator;
+
+  constructor(SUBSTRATE_URL: string) {
+    this.SUBSTRATE_URL = SUBSTRATE_URL;
+    this.calculator = new Calculator(new QueryClient(this.SUBSTRATE_URL));
+  }
+
+  async price() {
+    return await this.calculator.tftPrice();
+  }
+
+  @expose
+  @validateInput
+  async fromUSD(options: USDModel) {
```

Comment:
that's any object that implements `{amount: } `, I don't feel we need USDMdodel, but will see through the review 

Why didn't you add the return type? let's change the name to `convertUSDtoTFT`

Reply:
no need for async, make this code pure without side effect so either pass the price service or the price value and set it on the conversion service (and specify the direction that is tftInOneUSD for instance